ad7768-1fmcz/ad77681.c

```diff
@@ -211,30 +235,76 @@ uint8_t ad77681_get_rx_buf_len(struct ad77681_dev *dev)
 }
 
 /**
+ * Helper function to get the number of SPI bytes for INTERRUPT ADC DATA READ
+ * @param dev - The device structure.
+ * @return frame_bytes - the number of rx bytes
+ */
+uint8_t ad77681_get_frame_bytes(struct ad77681_dev *dev)
+{
+	uint8_t frame_bytes = 0;																// number of 8bit frames
+	
+	if (dev->conv_len == AD77681_CONV_24BIT)
+		frame_bytes += 3;
+	else
+		frame_bytes += 2;
+	
+	if (dev->crc_sel != AD77681_NO_CRC)
+		frame_bytes++;
+	
+	if (dev->status_bit)
+		frame_bytes++;
+	
+	if (frame_bytes % 2)																	// Conversion from number of 8bit frames to number of 16bit frames
+		frame_bytes = (frame_bytes / 2) + 1;
+	else
+		frame_bytes /= 2;
+	
+	dev->data_frame_bytes = frame_bytes;
+	
+	return frame_bytes;
+}
+
+/**
+ * This function reads the ADC data register by demand from the user in 8 bit frames and converts it to voltage
  * Read conversion result from device.
  * @param dev - The device structure.
  * @param adc_data - The conversion result data
+ * @param voltage  - Measured data transfered to voltage
  * @return 0 in case of success, negative error code otherwise.
  */
 int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
-				  uint8_t *adc_data)
+				uint8_t *adc_data,
+				double *voltage)
```

Comment:
This breaks the interface, other projects, that uses this function, wont compile(Ex:` ad77681_evb.c`)
A separate commit should be added with changes in `ad77681_evb.c`